contracts/wrapperContracts/WrapConversionRate.sol

```diff
@@ -0,0 +1,208 @@
+pragma solidity 0.4.18;
+
+
+import "../ERC20Interface.sol";
+import "../ConversionRates.sol";
+import "./WrapperBase.sol";
+
+
+contract WrapConversionRate is WrapperBase {
+
+    ConversionRates conversionRates;
+
+    //add token parameters
+    ERC20     addTokenToken;
+    uint      addTokenMinimalResolution; // can be roughly 1 cent
+    uint      addTokenMaxPerBlockImbalance; // in twei resolution
+    uint      addTokenMaxTotalImbalance;
+    uint      addTokenDataIndex;
+
+    //set token control info parameters.
+    ERC20[]     tokenInfoTokenList;
+    uint[]      tokenInfoPerBlockImbalance; // in twei resolution
+    uint[]      tokenInfoMaxTotalImbalance;
+    uint        tokenInfoDataIndex;
+
+    //valid duration
+    uint public pendingValidDurationBlocks;
+    uint        validDurationIndex;
+
+    //general functions
+    function WrapConversionRate(ConversionRates _conversionRates, address _admin) public
+        WrapperBase(PermissionGroups(address(_conversionRates)), _admin)
+    {
+        require (_conversionRates != address(0));
+        conversionRates = _conversionRates;
+        addTokenDataIndex = addDataInstance();
+        tokenInfoDataIndex = addDataInstance();
+        validDurationIndex = addDataInstance();
+    }
+
+    function addWrapperAsOperatorConversionRate() public onlyOperator {
+        conversionRates.addOperator(this);
+    }
+
+    // add token functions
+    //////////////////////
+    function setAddTokenData(ERC20 token, uint minimalRecordResolution, uint maxPerBlockImbalance, uint maxTotalImbalance) public onlyOperator {
+        require(minimalRecordResolution != 0);
+        require(maxPerBlockImbalance != 0);
+        require(maxTotalImbalance != 0);
+
+        //update data tracking
+        setNewData(addTokenDataIndex);
+
+        addTokenToken = token;
+        addTokenMinimalResolution = minimalRecordResolution; // can be roughly 1 cent
+        addTokenMaxPerBlockImbalance = maxPerBlockImbalance; // in twei resolution
+        addTokenMaxTotalImbalance = maxTotalImbalance;
+    }
+
+    function approveAddTokenData(uint nonce) public onlyOperator {
+        if(addSignature(addTokenDataIndex, nonce, msg.sender)) {
+            // can perform operation.
+            performAddToken();
+        }
+    }
+
+    function performAddToken() internal {
+        conversionRates.addToken(addTokenToken);
+
+        //token control info
+        conversionRates.setTokenControlInfo(
+            addTokenToken,
+            addTokenMinimalResolution,
+            addTokenMaxPerBlockImbalance,
+            addTokenMaxTotalImbalance
+        );
+
+        //step functions
+        int[] memory zeroArr = new int[](1);
+        zeroArr[0] = 0;
+
+        conversionRates.setQtyStepFunction(addTokenToken, zeroArr, zeroArr, zeroArr, zeroArr);
```

Comment:
I guess here you should set contract as operator. but you can set it as operator and remove it later.
anw, you seem to rely on calling set contract as operator first.